mock_transport/README.md

```diff
@@ -0,0 +1,47 @@
+# Mock testing framework
+
+## How to use
+
+### Writing examples or test cases
+
+If you want to write a test case or an example that supports the mock testing framework all you have to do is to call the `start_transaction` function before interacting with Azure. The function expects a transaction name that must be unique. For example:
+
+```rust
+azure_core::mock_transport::start_transaction("create_database_and_collection");
+```
+
+This call initializes the mock framework and it's all you need to add to your code (*note*: this call resolves in a noop if you do not enable the mock testing framework).
+
+#### Limitations
+
+* The mock testing framework supports async and multithreading but the call order **must** be deterministic. Also, for this same reason, tests cannot be run concurrently (with `cargo test -- --test-threads=1`).
+* The mock testing framework supports the pipeline architecture **only**. For this reason we cannot migrate the E2E test cases until the pipeline migration has been completed.
+
+### Recording an execution 
+
+1. Make sure the `AzureSDKforRustDumpPathRoot` environmental variable is set to a valid, writable path. This is where the requests/responses will be stored (into a subfolder named as the transaction name defined in the code).
```

Comment:
I've used the env variable because using relative paths forces the CI scripts to keep changing directories, but I agree, it doesn't warrant a more complex code. 👍.
 
I'll make it hardcoded and a subdirectory of each crate.